src/uu/stty/src/stty.rs

```diff
@@ -285,6 +360,72 @@ fn print_terminal_size(termios: &Termios, opts: &Options) -> nix::Result<()> {
     Ok(())
 }
 
+fn cc_to_index(option: &str) -> Option<SpecialCharacterIndices> {
+    for cc in CONTROL_CHARS {
+        if option == cc.0 {
+            return Some(cc.1);
+        }
+    }
+    None
+}
+
+// return Some(flag) if the input is a valid flag, None if not
+fn string_to_flag(option: &str) -> Option<AllFlags> {
+    // BSDs use a u32 for the baud rate, so any decimal number applies.
+    #[cfg(any(
+        target_os = "freebsd",
+        target_os = "dragonfly",
+        target_os = "ios",
+        target_os = "macos",
+        target_os = "netbsd",
+        target_os = "openbsd"
+    ))]
+    if let Ok(n) = option.parse::<u32>() {
+        return Some(AllFlags::Baud(n));
+    }
+
+    #[cfg(not(any(
+        target_os = "freebsd",
+        target_os = "dragonfly",
+        target_os = "ios",
+        target_os = "macos",
+        target_os = "netbsd",
+        target_os = "openbsd"
+    )))]
+    for (text, baud_rate) in BAUD_RATES {
+        if *text == option {
+            return Some(AllFlags::Baud(*baud_rate));
+        }
+    }
+
+    let (remove, name) = match option.strip_prefix('-') {
+        Some(s) => (true, s),
+        None => (false, option),
+    };
```

Comment:
I think you can simplify this with something like:
```rust
let remove = option.starts_with('-');
let name = option.trim_start_matches('-');
```
This assumes that `option` doesn't start with multiple `-`. Otherwise you could use `strip_prefix` with `unwrap_or_else`.